### PR TITLE
Implemented toggler component

### PR DESCRIPTION
**Pull request by mmsbrggr**
| Q | A
| --- | ---
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| License | MIT

#### What's in this PR?

The toggler frontend component, which can be used for requesting a boolean input from the user.

#### Example Usage

~~~javascript
initialState = {checked: false};
onChange = (checked) => setState({checked});
<Toggler checked={state.checked} onChange={onChange} />
~~~

### CODE REVIEW

**Comment by mmsbrggr** [comment]
@danrot This can be reviewed

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Toggler/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+The toggler is an boolean input element and has no internal state. It has to be managed from the outside,
+like shown in the following example:
+```
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/Toggler/README.md]: New line before code

**Comment by danrot** [comment]
@chirimoya What do you think about that? Should we copy all that code from the `Checkbox` to the `Toggler`, or should we build something like a `AbstractCheckbox` (should find a better naming for that...) that will be reused in both components?

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Radio/index.js [hunk 2]
```diff
@@ -1,4 +1,6 @@
 // @flow
 import Radio from './Radio';
+import RadioGroup from './RadioGroup';
 
 export default Radio;
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/Radio/index.js]: Not sure if we should still add a default export if the element is already in a named export... Haven't done it in the table component either... I think the important thing here would be consistency.

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Switch/README.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+The switch component is a general component with which checkboxes, radio buttons or toggler
+can be easily created. For the end user it is easier to one of those directly.
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/Switch/README.md]: `use one of these directly`

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Switch/Switch.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    handleChange = (event: SyntheticEvent<HTMLInputElement>) => {
+        if (this.props.onChange) {
+            this.props.onChange(event.currentTarget.checked, this.props.value);
+        }
+    };
+
+    render() {
+        const switchClasses = classNames(
+            switchStyles.switch,
+            this.props.className
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/Switch/Switch.js]: Use destructuring for all props

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Radio/Radio.js [hunk 5]
```diff
@@ -1,61 +1,42 @@
...
-                    <span className={radioStyles.labelText}>
-                        {children}
-                    </span>
-                }
-            </label>
+            <Switch
+                className={radioClass}
+                checked={this.props.checked}
+                value={this.props.value}
+                name={this.props.name}
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/Radio/Radio.js]: use destructuring

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Checkbox/Checkbox.js [hunk 6]
```diff
@@ -1,54 +1,37 @@
...
-                        <Icon className={checkboxStyles.icon} name="check" />
-                    }
-                </span>
-            </span>
+            <Switch
+                className={checkboxClass}
+                checked={this.props.checked}
+                value={this.props.value}
+                name={this.props.name}
+                icon={this.props.checked ? CHECKED_ICON : undefined}
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/Checkbox/Checkbox.js]: use destructuring

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Switch/tests/Switch.test.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    const component = render(<Switch checked={false} />);
+    expect(component).toMatchSnapshot();
+});
+
+test('The component should render in checked state', () => {
+    const component = render(<Switch checked={true} />);
+    expect(component).toMatchSnapshot();
+});
+
+test('The component should render in with class', () => {
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/Switch/tests/Switch.test.js]: no `in`

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Switch/tests/Switch.test.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    const component = render(<Switch className="my-class" checked={false} />);
+    expect(component).toMatchSnapshot();
+});
+
+test('The component should render in with type', () => {
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/Switch/tests/Switch.test.js]: no `in` and `with radio type`

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Toggler/Toggler.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+import togglerStyles from './toggler.scss';
+
+export default class Toggler extends React.PureComponent<SwitchProps> {
+    render() {
+        return (
+            <Switch
+                className={togglerStyles.toggler}
+                checked={this.props.checked}
+                value={this.props.value}
+                name={this.props.name}
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/Toggler/Toggler.js]: use destructuring

**Comment by danrot** [comment]
I also think it doesn't make any sense to define rules like `.checkbox, .checkbox.dark`... I would either just go with `.checkbox` and ignore the `.dark` class because it is the default, or create the `.checkbox` with the general definitions and nest `.dark` and `.light` to set the specific colors.

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Radio/index.js (hunk 2 above)
Comment (chirimoya) [reply src/Sulu/Bundle/AdminBundle/Resources/js/components/Radio/index.js]: remove default

**Pull request merged**